packages/react-components/src/components/chart/hooks/useDataZoomListeners.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+import { useViewport } from '../../../hooks/useViewport';
+import { EChartsType, getInstanceByDom } from 'echarts';
+
+export const useDataZoomListeners = (ref: React.RefObject<HTMLDivElement>) => {
+  const { setViewport } = useViewport();
+
+  useEffect(() => {
+    if (ref.current) {
+      const chart: EChartsType | undefined = getInstanceByDom(ref.current);
+
+      // known bug with zooming into live data https://github.com/apache/echarts/issues/11679
+      // this will prevent the data from moving if we are zoomed in
+      chart?.on('dataZoom', function () {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        //@ts-ignore
+        const currentDataZoom = chart?.getOption().dataZoom[0];
+
+        const newDataZoom = {};
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        //@ts-ignore
+        currentDataZoom.start > 0 ? (newDataZoom.startValue = currentDataZoom.startValue) : (newDataZoom.start = 0);
```

Comment:
should we move this to some other function which handles updating the dataZoom?

Reply:
that's basically what this whole file is for, just setting up event handlers for dataZoom